pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -49,8 +49,8 @@ const (
 type KubeNetworkInformation interface {
 	SetLocalPodDetails(owner, hostIP, podIP string, labels map[string]string)
```

Comment:
I'm wondering about splitting this interface. It seems this is only useful for the trace network (for network policy) generation gadget. I see that it's not used by others and it's even missing in tcpretrnas. 

Also, it seems to me `ContainerInfoGetters` is only required for this, so maybe we can change the logic a bit to not require all gadgets to implement `ContainerInfoGetters`, only the ones that have `SetLocalPodDetails`, by doing so we'll be able to enrich the endpoint information from the IP address regardless `ContainerInfoGetters`.

Reply:
Indeed `trace network` is the only gadget that needs that interface currently. I thought that we could create another operator just for gadgets that want to get network enrichment from pod namespace/name.

The cached container and service list which `KubeIPResolver` has can also be shared between the two operators. Or maybe the `KubeIPResolver` can check on two interfaces and selectively use them.
1. pod namespace/name -> network information
2. ip -> network information

Reply:
I agree that `SetLocalPodDetails` should be moved to a new interface that embeds `ContainerInfoGetters`. I would keep `ContainerInfoGetters` separated as it could be helpful for other operators in the future but for the kubeipresolver, ensure it is just required if the gadget also implements `SetLocalPodDetails`. Notice there is no verification right now. That's why tcpretrans works even without implementing it. 

I suggest something like this:

```go
// KubeNetworkInformation is for now a specific interface for `trace network` gadget
type KubeNetworkInformation interface {
	GetEndpoints() []*types.Endpoint
}

type AnotherFancyName interface {
	operators.ContainerInfoGetters
	SetLocalPodDetails(owner, hostIP, podIP string, labels map[string]string)
}
```

```go
func (k *KubeIPResolver) CanOperateOn(gadget gadgets.GadgetDesc) bool {
	km := kubemanager.KubeManager{}
	if !km.CanOperateOn(gadget) {
		return false
	}

	_, hasNetworkInf := gadget.EventPrototype().(KubeNetworkInformation)
	_, hasAnotherFancyInterface := gadget.EventPrototype().(AnotherFancyName)
	return hasNetworkInf || hasAnotherFancyInterface
}
```

```
func (m *KubeIPResolverInstance) enrich(ev any) {
	additionalInfo, enrichKubeNetInfo := ev.(KubeNetworkInformation)
	anotherFancyInfo, enrichAnotherFancyInfo := ev.(AnotherFancyName)

	[...]
	// Modify this function to do things based on enrichKubeNetInfo and
	// enrichAnotherFancyInfo
	[...]
```

If `enrich` becomes too complicated or unreadable, I would suggest dividing it into two functions and calling them from `EnrichEvent` based on `enrichKubeNetInfo` and `enrichAnotherFancyInfo`. Or we can consider having two operators as you suggested.